uber/model_checks.py

```diff
@@ -448,8 +449,19 @@ def valid_uses_allowed(promo_code):
 
 
 @validation.PromoCode
+def no_unlimited_free_badges(promo_code):
+    if promo_code.is_unlimited and (promo_code.is_free or (
+            promo_code.discount_type == PromoCode.FIXED_DISCOUNT and
+            promo_code.discount >= c.BADGE_PRICE)):
+        return 'Unlimited-use, free-badge promo codes are not allowed.'
```

Comment:
This is good, but will cause a possibly unwanted error in this scenario:
1. A promo code fixed discount of minus $50 is created.
2. The badge price at the time is $70 so the promo code can be made unlimited.
3. Later, for mysterious reasons, the badge price becomes $50.*
4. Someone tries to edit the promo code -- perhaps even to set the expiration date to the past -- and gets this error.

Obviously this is super, super, super unlikely, but can we limit this validation to only new promo codes? If you don't think we _should_ that's fine since this is not likely to be a problem, I just don't want us to be surprised by this happening.

*We did rolling badge prices last year where the badge price actually _decreased_ throughout the weekend, so it's not out of the realm of possibility.